### PR TITLE
[Improvement](topn) order by key topn query optimization

### PR DESCRIPTION
**Pull request by xiaokang**
# Proposed changes

Issue Number: close #xxx

## Problem summary

Describe your changes.

Optimize for order by key topn query like `SELECT * FROM table1 ORDER BY k1, k2 LIMIT n` in which k1 and k2 is the prefix of table sort key. 

This optimization is only for table with DUP_KEYS and UNQIUE_KEYS with merge on write.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [x] No
    - [x] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [x] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [x] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [x] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [x] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 36 seconds
 load time: 470 seconds
 storage size: 17122168178 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230105194532_clickbench_pr_74490.html

File: be/src/vec/olap/vcollect_iterator.h [hunk 1]
```diff
@@ -72,6 +72,26 @@ class VCollectIterator {
     }
 
 private:
+    // next for topn query
+    Status topn_next(Block* block);
```
Comment (Gabriel39) [inline be/src/vec/olap/vcollect_iterator.h:76]: _topn_next

File: be/src/vec/olap/vcollect_iterator.h [hunk 2]
```diff
@@ -72,6 +72,26 @@ class VCollectIterator {
...
+
+    class BlockRowposComparator {
```
Comment (Gabriel39) [inline be/src/vec/olap/vcollect_iterator.h:78]: BlockRowPosComparator

File: be/src/vec/olap/vcollect_iterator.cpp [hunk 3]
```diff
@@ -185,13 +195,170 @@ Status VCollectIterator::next(IteratorRowRef* ref) {
...
         return Status::Error<END_OF_FILE>();
     }
 }
 
+Status VCollectIterator::topn_next(Block* block) {
+    if (_topn_eof) {
+        return Status::Error<END_OF_FILE>();
+    }
+
+    auto cloneBlock = block->clone_without_columns();
```
Comment (Gabriel39) [inline be/src/vec/olap/vcollect_iterator.cpp:214]: clone_block

**Review by Gabriel39 — APPROVED** [review]
LGTM

**Pull request merged**